lib/generators/solidus_stripe/install/templates/config/initializers/solidus_stripe.rb

```diff
@@ -13,6 +13,32 @@
   # config.webhook_signature_tolerance = 150
 end
 
+# Allow sending the payment id in the payment form and avoid the automatic
+# creation of an additional payment by Spree::PaymentCreate.
+#
+# We're providing a inline module to make it easier to remove the mokey-patch
+# once the upstream PR is merged and generally available.
+#
+# Fixes https://github.com/solidusio/solidus/issues/2680
+# Fixed by https://github.com/solidusio/solidus/pull/4909
+begin
+  module SolidusStripe::PaymentCreateWithExistingPayment
+    def build
+      if attributes[:id]
+        @payment = order.payments.find(attributes[:id])
```

Comment:
It looks like we're still providing the if of an existing payment?

Reply:
We discussed a solution in a pairing review session: using the `payment_intent_id` instead. I think Elia will update the PR soon with that.